processor/configuration/src/main/java/com/redhat/service/smartevents/processor/sources/SourceResolver.java

```diff
@@ -0,0 +1,9 @@
+package com.redhat.service.smartevents.processor.sources;
+
+import com.redhat.service.smartevents.infra.models.actions.Action;
+import com.redhat.service.smartevents.infra.models.actions.Source;
+
+public interface SourceResolver extends SourceBean {
+
+    Action resolve(Source source, String customerId, String bridgeId, String processorId);
```

Comment:
@kostola Do we need to constrain the returned type here more? In reality this can only be an `Input` action in your model, we probably shouldn't allow people to return any type of Action 🤔